### PR TITLE
Add printing of layer legends

### PR DESCRIPTION
**Pull request by weskamm**
This enables printing of legends through the `geostyler-legend` module.
Currently, GeoJSON, WFS and WMS Layers are supported.

A new boolean `legend` has been introduced on the layer level of the spec to indicate that a legend should be generated when calling the new `createLegend` method.

In the future it may be better to have a separate legend configuration object with more control.

Examples and tests have been added.

@jahow please review

### CODE REVIEW

File: src/main/index.js
Comment (jahow) [inline src/main/index.js:194]: Maybe it would make sense here to put a warning in the console saying that no layers had legends enabled? Otherwise it might end up not doing anything and that could be confusing for the user

**Review by jahow — COMMENTED** [review]
This is looking great, thanks! are you waiting for the new release to include the `geostyler-legend` package as a dependency?

**Comment by weskamm** [comment]
> This is looking great, thanks! are you waiting for the new release to include the `geostyler-legend` package as a dependency?

Yes. Hope its ok?

**Comment by jahow** [comment]
> Yes. Hope its ok?

Absolutely!

**Comment by jahow** [comment]
I did a test and noticed that the crosses for the WFS layer do not show up (there's only the default OL style):

image

Not sure why yet, I'm taking a look

**Comment by jahow** [comment]
> Not sure why yet, I'm taking a look

So it's just a matter of version of the geostyler OL parser... could you maybe upgrade it to the latest version (either 2.5.0 or 3+ if it doesn't break anything)? Thanks! I'm approving the PR so that you can merge whenever it's ready, thanks again for the contribution :slightly_smiling_face:

**Review by jahow — APPROVED** [review]

**Comment by jahow** [comment]
Oh and also I allowed the PR to run so you might want to check what it says (let me know if you need pointers)

**Comment by weskamm** [comment]
> could you maybe upgrade it to the latest version (either 2.5.0 or 3+ if it doesn't break anything)?
yes right, i just upgraded to version 2.5.0 and the demo is working again.

regarding the typescript issues i am not sure how to proceed. As the project does not really uses typescript as its meant to be, i would just add some skips by adding e.g. `// @ts-nocheck` annotations.
If this is not ok please show me a better solution

**Comment by jahow** [comment]
> If this is not ok please show me a better solution

Typing is done using [JSDoc](https://jsdoc.app/) annotations. Let me try to fix the warnings on your branch.

**Comment by jahow** [comment]
@weskamm I think this PR can be merged, let me know if you're OK! the failure on the CI is related to permissions for the gh-pages deploy, I'll look into it later on.

**Comment by weskamm** [comment]
Thanks for the fixups, now i got the point. Yes, lets merge this!

**Pull request merged**
